packages/language/src/language-server/connection-handler.ts

```diff
@@ -91,5 +92,22 @@ export function startLanguageServer(connection: Connection): void {
       data: [],
     };
   });
+  connection.onDocumentHighlight((params) => {
+    const uri = params.textDocument.uri;
+    const position = params.position;
+    const textDocument = TextDocuments.get(uri);
+    const sourceFile = sourceFileHandler.getSourceFile(URI.parse(uri));
+    if (textDocument && sourceFile) {
+      const offset = textDocument.offsetAt(position);
+      const definition = referencesRequest(sourceFile, offset);
+      return definition.map((def) => {
+        return {
+          uri: def.uri,
+          range: rangeToLSP(textDocument, def.range),
+        };
+      });
```

Comment:
Suggestion: `uri` is not part of the expected return type `DocumentHighlight`. Also please make sure to filter based on the current URI. Otherwise, you will get document highlights spilling into the current file from other files.

Reply:
In fact, I would recommend to use `DocumentHighlight.create`, to ensure that everything is typed correctly (we should probably also do so for the other LSP services as well!)